File: interviews/blueorigin_1/python3/validate_input.py
def validateInput(command):
    max_command_len = 4
    valid_actions = ('L', 'R', 'F')

    pos_x = 0
    pos_y = 0
    orientation = 0

    if len(command) > max_command_len:
        return False

    for action in command:
        if action not in valid_actions:
            return False

    for i in range(4):
        for action in command:
            if action == 'L':
                orientation = (orientation - 90) % 360
            elif action == 'R':
                orientation = (orientation + 90) % 360
            elif action == 'F':
                if orientation == 0:
                    pos_y += 1
                elif orientation == 90:
                    pos_x += 1
                elif orientation == 180:
                    pos_y -= 1
                elif orientation == 270:
                    pos_x -= 1

    if pos_y == 0 and pos_x == 0:
        return True

    return False

if __name__ == '__main__':
    print('Enter a command: ', end=' ')
    command = input()
    print(validateInput(command))

File: interviews/blueorigin_1/c/validate_input.c
#include <stdio.h>
#include <stdbool.h>
#include <string.h>

bool validate_input(char* command);

int main(){
    char input[4];
    printf("Enter a command:  ");
    scanf("%s",input);
    printf("%s", validate_input(input) ? "True\n" : "False\n");
    return 0;
}

bool validate_input(char* command){
    static int max_command_len = 4;

    int len_command = strlen(command);
    int pos_x = 0;
    int pos_y = 0;
    int orientation = 0;

    //Validate command sequence length is not greater than max
    if(len_command> max_command_len)
    {
        return false;
    }

    //Validate command sequence contains valid actions
    //Validate command sequence to ensure return to origin
    for(int i = 0; i < 4*len_command; i++)
    {
        switch(command[i%len_command])
        {
            case 'L':
                orientation = (orientation + 270) % 360;
                break;
            case 'R':
                orientation = (orientation + 90) % 360;
                break;
            case 'F':
                switch(orientation)
                {
                    case 0:
                        pos_y++;
                        break;
                    case 90:
                        pos_x++;
                        break;
                    case 180:
                        pos_y--;
                        break;
                    case 270:
                        pos_x--;
                        break;
                }
                break;
            default :
                return false;
        }
    }

    if (pos_x == 0 && pos_y == 0){
        return true;
    }

    return false;
}


